// @ionic-super-tabs/angular/esm5/directives/proxies.js
import { __decorate } from "tslib";
/* tslint:disable */
/* auto-generated angular directive proxies */
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, EventEmitter, NgZone } from '@angular/core';
import { ProxyCmp, proxyOutputs } from './proxies-utils';
var SuperTab = /** @class */ (function () {
    function SuperTab(c, r, z) {
        this.z = z;
        c.detach();
        this.el = r.nativeElement;
    }
    SuperTab.ctorParameters = function () { return [
        { type: ChangeDetectorRef },
        { type: ElementRef },
        { type: NgZone }
    ]; };
    SuperTab = __decorate([
        ProxyCmp({ inputs: ['noScroll'], 'methods': ['getRootScrollableEl'] }),
        Component({ selector: 'super-tab', changeDetection: ChangeDetectionStrategy.OnPush, template: '<ng-content></ng-content>', inputs: ['noScroll'] })
    ], SuperTab);
    return SuperTab;
}());
export { SuperTab };
var SuperTabButton = /** @class */ (function () {
    function SuperTabButton(c, r, z) {
        this.z = z;
        c.detach();
        this.el = r.nativeElement;
    }
    SuperTabButton.ctorParameters = function () { return [
        { type: ChangeDetectorRef },
        { type: ElementRef },
        { type: NgZone }
    ]; };
    SuperTabButton = __decorate([
        ProxyCmp({ inputs: ['disabled'] }),
        Component({ selector: 'super-tab-button', changeDetection: ChangeDetectionStrategy.OnPush, template: '<ng-content></ng-content>', inputs: ['disabled'] })
    ], SuperTabButton);
    return SuperTabButton;
}());
export { SuperTabButton };
var SuperTabs = /** @class */ (function () {
    function SuperTabs(c, r, z) {
        this.z = z;
        c.detach();
        this.el = r.nativeElement;
        proxyOutputs(this, this.el, ['tabChange']);
    }
    SuperTabs.ctorParameters = function () { return [
        { type: ChangeDetectorRef },
        { type: ElementRef },
        { type: NgZone }
    ]; };
    SuperTabs = __decorate([
        ProxyCmp({ inputs: ['activeTabIndex', 'config'], 'methods': ['setConfig', 'selectTab'] }),
        Component({ selector: 'super-tabs', changeDetection: ChangeDetectionStrategy.OnPush, template: '<ng-content></ng-content>', inputs: ['activeTabIndex', 'config'] })
    ], SuperTabs);
    return SuperTabs;
}());
export { SuperTabs };
var SuperTabsContainer = /** @class */ (function () {
    function SuperTabsContainer(c, r, z) {
        this.z = z;
        c.detach();
        this.el = r.nativeElement;
        proxyOutputs(this, this.el, ['activeTabIndexChange', 'selectedTabIndexChange']);
    }
    SuperTabsContainer.ctorParameters = function () { return [
        { type: ChangeDetectorRef },
        { type: ElementRef },
        { type: NgZone }
    ]; };
    SuperTabsContainer = __decorate([
        ProxyCmp({ inputs: ['autoScrollTop', 'swipeEnabled'], 'methods': ['scrollToTop'] }),
        Component({ selector: 'super-tabs-container', changeDetection: ChangeDetectionStrategy.OnPush, template: '<ng-content></ng-content>', inputs: ['autoScrollTop', 'swipeEnabled'] })
    ], SuperTabsContainer);
    return SuperTabsContainer;
}());
export { SuperTabsContainer };
var SuperTabsToolbar = /** @class */ (function () {
    function SuperTabsToolbar(c, r, z) {
        this.z = z;
        c.detach();
        this.el = r.nativeElement;
        proxyOutputs(this, this.el, ['buttonClick']);
    }
    SuperTabsToolbar.ctorParameters = function () { return [
        { type: ChangeDetectorRef },
        { type: ElementRef },
        { type: NgZone }
    ]; };
    SuperTabsToolbar = __decorate([
        ProxyCmp({ inputs: ['color', 'scrollable', 'scrollablePadding', 'showIndicator'] }),
        Component({ selector: 'super-tabs-toolbar', changeDetection: ChangeDetectionStrategy.OnPush, template: '<ng-content></ng-content>', inputs: ['color', 'scrollable', 'scrollablePadding', 'showIndicator'] })
    ], SuperTabsToolbar);
    return SuperTabsToolbar;
}());
export { SuperTabsToolbar };
