// ngx-text-overflow-clamp/__ivy_ngcc__/dist/clamp.module.js
import { NgModule } from "@angular/core";
import { TextOverflowClampDirective } from "./clamp.directive";
import * as ɵngcc0 from '@angular/core';
var NgxTextOverflowClampModule = /** @class */ (function () {
    function NgxTextOverflowClampModule() {
    }
    /** @nocollapse */
    NgxTextOverflowClampModule.ctorParameters = function () { return []; };
NgxTextOverflowClampModule.ɵfac = function NgxTextOverflowClampModule_Factory(t) { return new (t || NgxTextOverflowClampModule)(); };
NgxTextOverflowClampModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: NgxTextOverflowClampModule });
NgxTextOverflowClampModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxTextOverflowClampModule, [{
        type: NgModule,
        args: [{
                imports: [],
                exports: [TextOverflowClampDirective],
                declarations: [TextOverflowClampDirective]
            }]
    }], function () { return []; }, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxTextOverflowClampModule, { declarations: function () { return [TextOverflowClampDirective]; }, exports: function () { return [TextOverflowClampDirective]; } }); })();
    return NgxTextOverflowClampModule;
}());
export { NgxTextOverflowClampModule };
